src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -978,6 +978,31 @@ static BOOL AttributeHasChangesOmittedQuality(MTRAttributePath * attributePath)
     return attributeValueToReturn;
 }
 
+- (nullable id)readAttributeWithEndpointID:(NSNumber *)endpointID
+                                 clusterID:(NSNumber *)clusterID
+                               attributeID:(NSNumber *)attributeID
+                                    params:(MTRReadParams * _Nullable)params
+                                     error:(NSError * __autoreleasing *)error
+{
+    NSDictionary<NSString *, id> * value = [self readAttributeWithEndpointID:endpointID
+                                                                   clusterID:clusterID
+                                                                 attributeID:attributeID
+                                                                      params:params];
+    if (value == nil) {
+        return nil;
+    }
+
+    auto * path = [MTRAttributePath attributePathWithEndpointID:endpointID clusterID:clusterID attributeID:attributeID];
+
+    auto * report = [[MTRAttributeReport alloc] initWithResponseValue:@{
```

Comment:
Nit: Maybe pull out that dictionary literal into a separate local variable... that's some exceptionally terrible clang-format indentation.

Reply:
Good idea.  Done.